corehq/apps/locations/bulk_management.py

```diff
@@ -336,13 +336,14 @@ def validate_and_parse_data_from_excel(self):
                 actual = set(sheet_reader.fieldnames) - set(optional_headers)
                 expected = set(LOCATION_SHEET_HEADERS_BASE.values())
                 if actual != expected:
+                    if expected - actual:
+                        detail = "'{}' are missing.".format(", ".join(expected - actual))
+                    else:
+                        detail = "'{}' are not recognized.".format(", ".join(actual - expected))
```

Comment:
why not keep both?
also, do we want these messages to be translated?

Reply:
Translated in f4c9c8733a8. What do you mean about keeping both?

Reply:
missing and unrecognized.

Reply:
+1 
I think Manish means this:
```python
if expected - actual:
    detail = "'{}' are missing.".format(", ".join(expected - actual))
if actual - expected:
    detail = "'{}' are not recognized.".format(", ".join(actual - expected))
```

Reply:
Got it, also spoke with Manish. Added 24a1dffc29c to fix this up.